packages/storage/src/common/utils.ts

```diff
@@ -114,16 +114,16 @@ export function replaceSchemeWithGatewayUrl(
   const scheme = Object.keys(gatewayUrls).find((s) => uri.startsWith(s));
   const schemeGatewayUrls = scheme ? gatewayUrls[scheme] : [];
 
-  if (!scheme) {
-    return uri;
-  }
-
-  if (index > schemeGatewayUrls.length) {
+  if ((!scheme && index > 0) || index >= schemeGatewayUrls.length) {
     throw new Error(
       "[GATEWAY_URL_ERROR] Failed to resolve gateway URL - ran out of gateway URLs to try.",
```

Comment:
what if you just return undefined here instead of throwing and re-catching your own error?